### ISSUE TITLE
Follow OOP design principle for `CarController`

### ISSUE THREAD

**Issue by david-eom** [Team-Aston-Martin-Henz/Cyberacer#14]
The current implementation of `CarController` is too cluttered, and an AI car does not require so many variables and methods.
Maybe it is time to consider following OOP design principle, and have one abstract `CarController` class, with `PlayerCarController` and `AICarController` inheriting from that.
Needs to read up more on how inheritance works for Unity C# Script.

**Comment by Timothyoung97** [comment]
Currently, I am thinking if we analyse from the coding practice and principle point of view, this part shall be placed at milestone 3 to rank up in software evaluation / qualitative analysis.

From a gameplay point of view, as of now, the CarController was developed with the AI Car component and PlayerCar component clustered together. This is because in the future when we create a new scene, all AI cars will be generated randomly. And all it takes is to trigger the AI CarController's `isAI` boolean to be true and we can trigger all the AI property. This, I think is rather easy on our side, and it reduces a lot of problems that may be created after we refactor the code when we are still developing.

As of now, I think our priority should set to focus on the completion of features -> then we proceed to enhance the script

**Comment by david-eom** [comment]
Yep, I agree that the priority should be the completion of features, I go about resolving this issue probably towards the end of the month. I still think that it might be necessary, and cluttered / repeated codes might not be optimal for the future development of the game e.g. weapon system (we have to have two sets of code for AI and player once again). But I do see where you're coming from, and will not spend too much time on this before milestone 3.
